src/dns/rfc1035.cc

```diff
@@ -522,11 +522,11 @@ rfc1035QueryUnpack(const char *buf, size_t sz, unsigned int *off, rfc1035_query
         memset(query, '\0', sizeof(*query));
         return 1;
     }
-    memcpy(&s, buf + *off, 2);
-    *off += 2;
+    memcpy(&s, buf + *off, sizeof(s));
+    *off += sizeof(s);
```

Comment:
We should not do this until `s` is guaranteed to be exactly 2 bytes in size. Technically, `unsigned short` does not provide that guarantee... Would you like to change `s` type to `uint16_t`? rfc1035RRPack() already uses fixed-size types.

Reply:
sure, done